json-schema-generator/src/main/java/io/micronaut/jsonschema/generator/utils/FileProcessor.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.jsonschema.generator.utils;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.micronaut.jsonschema.model.Schema;
+import io.micronaut.jsonschema.serialization.JsonSchemaMapperFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+public class FileProcessor {
+
+    // configurable by user
+    private static List<String> allowedUrlPatterns = List.of(
+        "^https://.*/.*.json"
+    );
+
+    public static Schema getJsonSchema(SourceGeneratorConfig config) {
+        ObjectMapper jsonMapper = JsonSchemaMapperFactory.createMapper();
+        try {
+            if ((config.jsonUrl() != null && !config.jsonUrl().isBlank()) || config.inputStream() != null) {
+                var inputStream = config.inputStream();
+                if (inputStream == null) {
+                    inputStream = downloadAsStream(config.jsonUrl());
+                }
+                String jsonString = new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);
+                return jsonMapper.readValue(jsonString, Schema.class);
+            } else if (config.jsonFile() != null) {
+                return jsonMapper.readValue(config.jsonFile(), Schema.class);
+            } else {
+                throw new RuntimeException("Missing required config.jsonUrl() or config.inputStream()");
+            }
```

Comment:
I would like to see this made extensible. I can see a use case whereby you may want to query oracle database (or some other database) for the schema and generate it from there https://docs.oracle.com/en/database/oracle/oracle-database/23/adjsn/json-schema.html#GUID-7D67DF73-CB21-4878-AF93-D1A213411EC0